### ISSUE TITLE
Collecting Data

### ISSUE THREAD

**Issue by GaelGil** [GaelGil/notebooks#3]
In order to have a good model I need to have lots of examples. In my case I need a lot of images of sneakers for my sneaker classifier [project](https://github.com/GaelGil/notebooks/tree/master/turicreate). I want to be able to label some sneaker models but all models can have multiple colors. For example `jordan_one` is a model and it can come in many colors. If all my current images of the `jordan_one` model are in red at different angles, would that affect labeling a `jordan_one` in another color or would that not matter. Should I be collecting Images of all angles and all colors?

**Comment by camoverride** [comment]
Good question. A different way of phrasing your question is "will my model work in the wild?" The answer is: *it depends.*

Let's say you collect 10,000 images of shoes, divided evenly into 4 categories (2,500 images in each category). These images are of different sizes, resolutions, and angles... some images might even be messy (maybe there's dirt on the shoe), or something occluding the show. The way a typical data science project works is that you then divide these 10,000 images into a training and a test set (maybe 8,000 in training and 2,000 in the test set). Then you train the model on the training set and test its performance on the test set. Your concern (which is valid) is what happens when you take a totally new picture of a show that was neither in the training or the test set and show it to the model...

If the same "function" you used to generate the training/test data is used to generate the new shoe image, then you should expect the model's performance to resemble the performance on the test set. The *generating function* might be something like snapping a picture of a shoe with your phone, or taking a shoe picture from Google Images. The *generating function* defines the data that gets generated, and if the new data you show your model is generated in an entirely different way, you should expect your model's performance to be worse.

That being said, I found a database of faces where each face picture was labeled with one of 7 emotions (happy, sad, angry, etc). I trained a deep learning model that got good performance (about 80% accuracy -- in comparison, a random quess would yield 14% accuracy). Then I took my webcam and pointed it at my face, took pictures, and had my model try and predict my emotions -- the model's performance was worse than on the test data. Why? Probably because the lighting in my apartment was different than the lighting used when the training pictures were taken -- different generating functions were used for my webcam images and the training data.

So what can you do about it?

Option 1: data augmentation. Just take the data you have and mutate it in different ways so that you have more data. The simplest way of doing this is to just take a mirror reflection of the image  -- after all, a shoe is still a shoe regardless of which direction its facing: read this https://machinelearningmastery.com/how-to-configure-image-data-augmentation-when-training-deep-learning-neural-networks/

Option 2: different models. If you're concerned that your model might have only trained on blue shoes but you want it to recognize red shoes as well, then use a model that removes color information (or preprocess your images so that they're black and white). This might make it seem like you're "losing data" (and you are) but sometimes this is a good thing, because it prevents your models from overfitting.

Option 3: stop caring. Just train your models, see how well they perform, and maybe the color errors are small and infrequent and don't really matter. Data scientists also have to examine their model errors to see whether they matter.

**Comment by GaelGil** [comment]
this notebook does not exist anymore
